src/test/java/com/chikli/hudson/plugin/naginator/NaginatorScheduleActionTest.java

```diff
@@ -0,0 +1,335 @@
+import hudson.matrix.MatrixRun;
+import hudson.matrix.MatrixBuild;
+import hudson.matrix.MatrixProject;
+import hudson.model.AbstractBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.BuildListener;
+import hudson.model.FreeStyleBuild;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.Builder;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+/**
+ *
+ */
+public class NaginatorScheduleActionTest {
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+    
+    public static class ScheduleActionBuilder extends Builder {
+        private final NaginatorScheduleAction[] actions;
+        
+        public ScheduleActionBuilder(NaginatorScheduleAction... actions) {
+            this.actions = actions;
+        }
+        
+        @Override
+        public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener)
+                throws InterruptedException, IOException {
+            if (build instanceof MatrixRun) {
+                MatrixBuild parent = ((MatrixRun) build).getParentBuild();
+                if (parent.getAction(NaginatorScheduleAction.class) == null) {
+                    for (NaginatorScheduleAction action : actions) {
+                        parent.addAction(action);
+                    }
+                }
+            } else {
+                for (NaginatorScheduleAction action : actions) {
+                    build.addAction(action);
+                }
+            }
+            return true;
+        }
+    }
+    
+    private static class AlwaysFalseScheduleAction extends NaginatorScheduleAction {
+        public AlwaysFalseScheduleAction(int maxSchedule, ScheduleDelay delay, boolean rerunMatrixPart) {
+            super(maxSchedule, delay, rerunMatrixPart);
+        }
+        
+        @Override
+        public boolean shouldSchedule(Run<?, ?> run, TaskListener listener, int retryCount) {
+            return false;
+        }
+    }
+    
+    private static class MatrixConfigurationScheduleAction extends NaginatorScheduleAction {
+        private final String combinationFilter;
+        
+        public MatrixConfigurationScheduleAction(String combinationFilter, int maxSchedule, ScheduleDelay delay, boolean rerunMatrixPart) {
+            super(maxSchedule, delay, rerunMatrixPart);
+            this.combinationFilter = combinationFilter;
+        }
+        
+        @Override
+        public boolean shouldScheduleForMatrixRun(MatrixRun run, TaskListener listener) {
+            return run.getParent().getCombination().evalGroovyExpression(
+                    run.getParent().getParent().getAxes(),
+                    combinationFilter
+            );
+        }
+    }
+    
+    @Test
+    public void testShouldReschedule() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+        FreeStyleBuild b = p.scheduleBuild2(0).get();
+        
+        {
```

Comment:
Why do you need anonymous block code?
You can just separate lines into groups of lines with new line separator or add some comment before group of lines